libs/shared/src/shared.service.ts

```diff
@@ -1,4 +1,33 @@
 import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { RmqContext } from '@nestjs/microservices';
+import { RmqOptions, Transport } from '@nestjs/microservices';
 
 @Injectable()
-export class SharedService {}
+export class SharedService {
+  constructor(private readonly configService: ConfigService) {}
+
+  getRmqOptions(queue: string): RmqOptions {
+    const USER = this.configService.get('RABBITMQ_USER');
+    const PASSWORD = this.configService.get('RABBITMQ_PASSWORD');
+    const HOST = this.configService.get('RABBITMQ_HOST');
+    this.configService.get(`RABBITMQ_${queue}_QUEUE`);
+    return {
+      transport: Transport.RMQ,
+      options: {
+        urls: [`amqp://${USER}:${PASSWORD}@${HOST}`],
+        queue,
+        noAck: false,
+        queueOptions: {
+          durable: true,
+        },
+      },
+    };
+  }
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Long method with multiple responsibilities</b></div>
<div id="fix">

The `getRmqOptions` method is quite long and handles multiple responsibilities. Consider breaking it down into smaller, more focused methods for better maintainability.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -10,17 +10,22 @@
    getRmqOptions(queue: string): RmqOptions {
      const USER = this.configService.get('RABBITMQ_USER');
      const PASSWORD = this.configService.get('RABBITMQ_PASSWORD');
      const HOST = this.configService.get('RABBITMQ_HOST');
 -    this.configService.get('RABBITMQ_${queue}_QUEUE');
 +    const QUEUE = this.configService.get('RABBITMQ_${queue}_QUEUE');
 +    return this.createRmqOptions(USER, PASSWORD, HOST, QUEUE || queue);
 +  }
 +
 +  private createRmqOptions(user: string, password: string, host: string, queue: string): RmqOptions {
      return {
        transport: Transport.RMQ,
        options: {
 -        urls: ['amqp://${USER}:${PASSWORD}@${HOST}'],
 +        urls: ['amqp://${user}:${password}@${host}'],
          queue,
          noAck: false,
          queueOptions: {
            durable: true,
          },
        },
      };
    }
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/omkardongre/messenger-api/pull/6#issuecomment-2448418554>#48ef5b</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged